cupy/core/core.pyx

```diff
@@ -2890,6 +2901,80 @@ cpdef ndarray tensordot_core(
     return ret
 
 
+cpdef ndarray tensordot_core_v11(
+        Py_ssize_t transa, Py_ssize_t transb, Py_ssize_t m, Py_ssize_t n,
+        Py_ssize_t k, ndarray a, Py_ssize_t lda, ndarray b, Py_ssize_t ldb,
+        ndarray c, Py_ssize_t ldc):
+    cdef float one_f, zero_f
+    cdef double one_d, zero_d
+    cdef cuComplex one_F, zero_F
+    cdef cuDoubleComplex one_D, zero_D
+
+    if c.dtype.char in 'efF':
+        compute_type = cublas.CUBLAS_COMPUTE_32F
+    elif c.dtype.char in 'dD':
+        compute_type = cublas.CUBLAS_COMPUTE_64F
+
+    compute_capability = int(device.get_compute_capability())
+    algo = cublas.CUBLAS_GEMM_DEFAULT
+    if ((compute_capability >= 80) or
+            (compute_capability >= 70 and c.dtype == 'e')):
+        algo = cublas.CUBLAS_GEMM_DEFAULT_TENSOR_OP
+
+    a_cuda_dtype = dtype_to_cuda_dtype(a.dtype, is_half_allowed=True)
+    b_cuda_dtype = dtype_to_cuda_dtype(b.dtype, is_half_allowed=True)
+    c_cuda_dtype = dtype_to_cuda_dtype(c.dtype, is_half_allowed=True)
+    handle = device.get_cublas_handle()
```

Comment:
Note: `get_cublas_handle()` currently returns `size_t`, but the `gemmEx` routine requires `intptr_t`. I will send a PR to fix `get_*_handle()`.

Reply:
See #3746.